test base

hwjsnc opened the pull request:
Added a starting point for tests. Only MasterQuartzBase for now.

Ideally, github actions just need to `sh run_tests.sh`.

The commit adds a submodule with the kaitai python runtime.

Windows might need an alternate solution for the symbolic link to `kaitaistruct.py`.

ghost commented:
I think `main.yml` should be renamed to `ci.yml`  main doesn't have any meaning here

hwjsnc committed:
CI setup

hwjsnc commented:
- renamed to `ci.yml`
- removed the symbolic link and shell script for windows compat
- uses `unittest` now
- slight changes to the yml (e.g. replaced `cd` with `working-directory`)
- properly pinned the ksc binary
- restructured to test all games at the same time (once we get there)
